packages/common/src/ui/components/modals/ConfirmationModal/ConfirmationModalProvider.tsx

```diff
@@ -0,0 +1,104 @@
+import React, {
+  FC,
+  useCallback,
+  useMemo,
+  useContext,
+  useState,
+  createContext,
+} from 'react';
+import { ConfirmationModal } from './ConfirmationModal';
+
+interface ConfirmationModalState {
+  open: boolean;
+  message: string;
+  title: string;
+  iconType?: 'alert' | 'info';
+  onConfirm: (() => void) | (() => Promise<void>);
+}
+
+interface ConfirmationModalControllerState extends ConfirmationModalState {
+  setState: (state: ConfirmationModalState) => void;
+  setMessage: (message: string) => void;
+  setTitle: (title: string) => void;
+  setOnConfirm: (onConfirm: (() => Promise<void>) | (() => void)) => void;
+  setOpen: (open: boolean) => void;
+}
+
+const Context = createContext<ConfirmationModalControllerState>({
+  open: false,
+  message: '',
+  title: '',
+  iconType: 'info',
+  onConfirm: () => {},
+  setState: () => {},
+  setMessage: () => {},
+  setTitle: () => {},
+  setOnConfirm: () => {},
+  setOpen: () => {},
+});
+
+export const ConfirmationModalProvider: FC = ({ children }) => {
+  const [confirmationModalState, setState] = useState<ConfirmationModalState>({
+    open: false,
+    message: '',
+    title: '',
+    iconType: 'info',
+    onConfirm: async () => {},
+  });
+  const { open, message, title, iconType, onConfirm } = confirmationModalState;
+
+  const confirmationModalController: ConfirmationModalControllerState = useMemo(
+    () => ({
+      setMessage: (message: string) =>
+        setState(state => ({ ...state, message })),
+      setTitle: (title: string) => setState(state => ({ ...state, title })),
+      setOnConfirm: (onConfirm: (() => Promise<void>) | (() => void)) =>
+        setState(state => ({ ...state, onConfirm })),
+      setOpen: (open: boolean) => setState(state => ({ ...state, open })),
+      setState,
+      ...confirmationModalState,
+    }),
+    [setState, confirmationModalState]
+  );
+
+  return (
+    <Context.Provider value={confirmationModalController}>
+      {children}
+      <ConfirmationModal
```

Comment:
technically this is hitting against the general context problem of state changing this far up the tree. it doesn't seem to have a performance hit - profiling doesn't show anyheavy renders, all below 10ms so it seems fine..